istioctl/pkg/waypoint/waypoint.go

```diff
@@ -102,6 +102,9 @@ func Cmd(ctx cli.Context) *cobra.Command {
 			res := strings.ReplaceAll(string(b), `  creationTimestamp: null
 `, "")
 			res = strings.ReplaceAll(res, `status: {}
+`, "")
+			// stripping infrastructure introduced in Gateway v1 if empty
+			res = strings.ReplaceAll(res, `  infrastructure: {}
```

Comment:
@hanxiaop but we don't know here if the CRDs users have are v1 or before.
The infrastructure field is generated by the Gateway instance we create.
P.S. we also don't know which CRD version will include this field in the scheme because the most recent one still doesn't have it.